automation/taskcluster/lib/tasks.py

```diff
@@ -342,61 +314,105 @@ def _craft_default_task_definition(
             },
         }
 
-    def craft_signing_task(
-        self, assemble_task_id, variant=None, is_staging=True,
-    ):
-        is_nightly = variant is None    # We currently build all variants in nightly
-        signing_format = 'autograph_apk_reference_browser'
-        payload = {
-            "upstreamArtifacts": [{
-                "paths": _APKS_PATHS if is_nightly else [DEFAULT_APK_ARTIFACT_LOCATION],
-                "formats": [signing_format],
-                "taskId": assemble_task_id,
-                "taskType": "build",
-            }],
-        }
+    def craft_master_commit_signing_task(self, assemble_task_id, variant):
+        architecture, build_type, product = \
+            get_architecture_and_build_type_and_product_from_variant(variant)
+        product = convert_camel_case_into_kebab_case(product)
+        postfix = convert_camel_case_into_kebab_case('{}-{}'.format(architecture, build_type))
 
-        return self._craft_default_task_definition(
-            worker_type='mobile-signing-dep-v1' if is_staging else 'mobile-signing-v1',
-            provisioner_id='scriptworker-prov-v1',
-            name="sign: {}".format(variant if variant else 'all'),
-            description="Sign builds",
-            payload=payload,
-            dependencies=[assemble_task_id],
-            routes=self._craft_nightly_routes(is_nightly, is_staging),
-            scopes=[
-                "project:mobile:reference-browser:releng:signing:format:{}".format(signing_format),
-                "project:mobile:reference-browser:releng:signing:cert:{}".format(
-                    'dep-signing' if is_staging or not is_nightly else 'release-signing'
-                )
-            ],
+        routes = []
+        if self.repo_url == _OFFICIAL_REPO_URL and self.short_head_branch == 'master':
```

Comment:
It might be worth moving the `self.short_head_branch == "master"` logic up so that `craft_master_commit_signing_task` is only called if the branch is "master" (or renaming the method)

Reply:
True! Fixed.